src/protocol/prf_sharding/mod.rs

```diff
@@ -480,6 +604,49 @@ where
     ))
 }
 
+/// If the `attribution_window_seconds` is not `None`, we calculate the time
+/// difference between the trigger event and the most recent source event, and
+/// returns a secret-shared bit indicating if the trigger event is within the
+/// attribution window.
+async fn is_trigger_event_within_attribution_window<C>(
+    ctx: C,
+    record_id: RecordId,
+    attribution_window_seconds: Option<NonZeroU32>,
+    trigger_event_timestamp: &BitDecomposed<Replicated<Gf2>>,
+    source_event_timestamp: &BitDecomposed<Replicated<Gf2>>,
+) -> Result<Replicated<Gf2>, Error>
+where
+    C: UpgradedContext<Gf2, Share = Replicated<Gf2>>,
+{
+    // if there is no attribution window, then all trigger events are attributed
+    if attribution_window_seconds.is_none() {
+        let share_of_one = Replicated::share_known_value(&ctx, Gf2::ONE);
+        return Ok(share_of_one);
+    }
```

Comment:
It's unfortunate that we are still going to multiply this sharing of ONE with the "did trigger get attributed" in the case of no windowing.